crates/core/src/db/models.rs

```diff
@@ -20,6 +20,132 @@ use crate::db::custom_types::{
     VoteThresholdType, VoteTippingEnum, VoteWeightV1, VoteWeightV1Enum, WhitelistMintMode,
 };
 
+/* MPL LISTING REWARDS */
+
+/// A row in the `rewards_purchase_tickets` table
+#[derive(Debug, Clone, Queryable, Insertable, AsChangeset, Associations)]
+#[diesel(treat_none_as_null = true)]
+pub struct RewardsPurchaseTicket<'a> {
+    /// The address of this account
+    pub address: Cow<'a, str>,
+    /// reward center associated of the purchase ticket
+    pub reward_center_address: Cow<'a, str>,
+    /// the buyer of the nft
+    pub buyer: Cow<'a, str>,
+    /// the seller of the nft
+    pub seller: Cow<'a, str>,
+    /// the metadata of the nft purchased
+    pub metadata: Cow<'a, str>,
+    /// price of the nft
+    pub price: i64,
+    /// number of tokens sold
+    pub token_size: i64,
+    /// the date and time of the purchase
+    pub created_at: NaiveDateTime,
+    /// The slot number of the most recent update for this account
+    pub slot: i64,
+    /// The write version of the most recent update for this account
+    pub write_version: i64,
+}
+
+/// A row in the `reward_centers` table
+#[derive(Debug, Clone, Queryable, Insertable, AsChangeset, Associations)]
+#[diesel(treat_none_as_null = true)]
+pub struct RewardCenter<'a> {
+    /// The address of this account
+    pub address: Cow<'a, str>,
+    /// the mint of the token used as rewards
+    pub token_mint: Cow<'a, str>,
+    /// the auction house associated to the reward center
+    pub auction_house: Cow<'a, str>,
+    /// the bump of the pda
+    pub bump: i16,
+    /// The slot number of the most recent update for this account
+    pub slot: i64,
+    /// The write version of the most recent update for this account
+    pub write_version: i64,
+}
+
+/// A row in the `rewards` table
+#[derive(Debug, Clone, Queryable, Insertable, AsChangeset, Associations)]
+#[diesel(treat_none_as_null = true)]
+#[belongs_to(parent = "RewardCenter<'_>", foreign_key = "reward_center_address")]
+pub struct ListingRewardRule<'a> {
+    /// The reward_center being created
+    pub reward_center_address: Cow<'a, str>,
+    /// Basis Points to determine reward ratio for seller
+    pub seller_reward_payout_basis_points: i16,
+    /// Payout Divider for determining reward distribution to seller/buyer
+    pub payout_divider: i16,
+}
```

Comment:
@imabdulbasit can you update saving of the reward rules to be on the reward_center record? Its an object on the reward center so not being processed at a different time. 

```
reward_centers
----------------
reward_rules_numeral
reward_rules_operand
reward_rules_seller_reward_payout_basis_points
```

There has been slight modifications to the reward rules so don't start on this until I get you an update crate to use for deserialization.

Reply:
@imabdulbasit new package uploaded to crate.io.

I small name change from mpl-reward-center to hpl-reward-center (holaplex program library)